agent/config_endpoint.go

```diff
@@ -32,6 +32,14 @@ func (s *HTTPServer) configGet(resp http.ResponseWriter, req *http.Request) (int
 	}
 	pathArgs := strings.SplitN(strings.TrimPrefix(req.URL.Path, "/v1/config/"), "/", 2)
 
+	if len(pathArgs) == 2 {
```

Comment:
This is complicated. We are using one handler for endpoints like

```
/v1/config/proxy-defaults/global # read that single config entry
/v1/config/service-resolver      # list all the service resolvers
```

In this case we only allow wildcarding the namespace query parameter/header if its a listing operations (path length == 1). Otherwise we need to disallow the wildcard.

Personally I would love to see us move away from our current router/handler infrastructure and move to something that is more expressive with regards to the HTTP URLs. That way `/v1/config/[^/]+` and `/v1/config/[^/]+/[^/]+` could route to two separate handlers and we could remove the branching from the method all together.

For the time being I have added a follow on task to refactor this and put the parsing into the switch cases as they should be fine going in there.

Reply:
Agree on more expressive route matchers -- thanks for looking into this!